### PR TITLE
[AS] ADY-17 awaiting/invalid data emails for UBOs

### PR DESCRIPTION
**Pull request by asamrae2y**

### CODE REVIEW

File: src/test/java/com/adyen/mirakl/cucumber/stepdefs/helpers/stepshelper/StepDefsHelper.java [hunk 1]
```diff
@@ -26,6 +30,9 @@
 
 import static org.awaitility.Awaitility.await;
 
+@WebAppConfiguration
+@SpringBootTest
+@ContextConfiguration(classes = AdyenMiraklConnectorApp.class)
```
Comment (pandemicGander) [inline src/test/java/com/adyen/mirakl/cucumber/stepdefs/helpers/stepshelper/StepDefsHelper.java]: reason we've added this now?

File: src/test/features/adyenemailnotifications/kyccheckemail.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Feature: KYC check intervention emails
+
+    @ADY-17
+    Scenario: Remedial email is sent to seller upon Identity Check awaiting data notification
+        Given a new Business shop has been created in Mirakl without mandatory Shareholder Information
```
Comment (rmcotton) [inline src/test/features/adyenemailnotifications/kyccheckemail.feature]: for me the words should be, only partial shareholder data supplied with some mandatory information missing.

but keep in mind companies may have only 1 share holder, so we need to be sure it doesnt always send 4 emails.

File: src/test/java/com/adyen/mirakl/cucumber/stepdefs/MiraklAdyenSteps.java [hunk 3]
```diff
@@ -565,20 +569,60 @@ public void theUpdatedDocumentsAreSuccessfullyUploadedToAdyen(DataTable table) t
...
-                    .assertThat(documents)
-                    .hasSize(2);
+            Assertions
+                .assertThat(documents)
+                .hasSize(2);
         }
     }
+
+    @Then("^a remedial email will be sent for each ubo$")
+    public void aRemedialEmailWillBeSentForEachUbo(String title) throws Throwable {
```
Comment (rmcotton) [inline src/test/java/com/adyen/mirakl/cucumber/stepdefs/MiraklAdyenSteps.java]: wheres the test for only some UBO's. lets say we have 3 UBO's, 1 is setup correct, 2 missing some mandatory data. we should only get an email for the ubo's missing mandatory data.

File: src/test/java/com/adyen/mirakl/cucumber/stepdefs/MiraklAdyenSteps.java [hunk 4]
```diff
@@ -100,11 +101,24 @@ public void aNewShopHasBeenCreatedAsABusinessWithUBOData(String legalEntity, Dat
...
 
-    @Given("^a new (.*) shop has been created in Mirakl without mandatory Shareholder Information$")
-    public void aNewBusinessShopHasBeenCreatedInMiraklWithoutMandatoryShareholderInformation(String legalEntity, DataTable table) throws Throwable {
+    @Given("^a new (.*) shop has been created in Mirakl with some Mandatory data missing$")
+    public void aNewBusinessShopHasBeenCreatedInMiraklWithoutMandatoryShareholderInformation(String legalEntity, DataTable table) {
         List<Map<String, String>> cucumberTable = table.getTableConverter().toMaps(table, String.class, String.class);
         MiraklCreatedShops shops = miraklShopApi.createBusinessShopWithMissingUboInfo(miraklMarketplacePlatformOperatorApiClient, cucumberTable, legalEntity);
         shop = retrieveCreatedShop(shops);
+        //  used for scenario (ADY-17)
+        if (scenario.getSourceTagNames().contains("@ADY-17")) {
```
Comment (rmcotton) [inline src/test/java/com/adyen/mirakl/cucumber/stepdefs/MiraklAdyenSteps.java]: this is a bit nasty to ship as source code. it'll be a complete magic number to anyone outside scope of the project

Comment (asamrae2y) [reply src/test/java/com/adyen/mirakl/cucumber/stepdefs/MiraklAdyenSteps.java]: agreed. i'll remove dependency for tags in this instance and in a separate commit will have a better way of tags.

**Pull request merged**